pkg/collector/corechecks/servicediscovery/module/comm_test.go

```diff
@@ -124,9 +133,78 @@ func TestShouldIgnoreComm(t *testing.T) {
 			require.EventuallyWithT(t, func(collect *assert.CollectT) {
 				proc, err := customNewProcess(int32(cmd.Process.Pid))
 				require.NoError(t, err)
-				ignore := shouldIgnoreComm(proc)
+
+				ignore := discovery.shouldIgnoreComm(proc)
 				require.Equal(t, test.ignore, ignore)
-			}, 30*time.Second, 100*time.Millisecond)
+			}, 10*time.Second, 100*time.Millisecond)
+		})
+	}
+}
+
+type ignoredCommTestAttr struct {
+	name  string   // The name of the test.
+	comms []string // list of command names to test
+}
+
+func (a *ignoredCommTestAttr) buildIgnoredCommandsString() string {
+	var builder strings.Builder
+	for _, cmd := range a.comms {
+		builder.WriteString(cmd)
+		builder.WriteString("   ") // intentionally multiple spaces for sensitivity testing
+	}
+	return builder.String()
+}
```

Comment:
1. `return strings.Join("    ", a.comms)`
2. shouldn't it be comma separated?
3. why declaring `ignoredCommTestAttr` as a struct? why can't `buildIgnoredCommandString` be a simple function?

Reply:
It turns out cfg.BindEnvAndSetDefault(..., []string) creates list of elements separated by spaces in configuration parameter. When I set comma separator, then each comma is treated as element.

Reply:
I've replaced this function with simple "join"